### DISCUSSION TITLE
canView() Method to Control Page Access

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3629] [Ideas]
Widgets currently have a `canView()` method so that you can easily control whether a widget should be visible or not. It would be great if we could have something similar for pages. While I can currently remove the page from the navigation easily enough, I cannot actually prevent someone from directly accessing the page URL itself.

I know that [this package](https://github.com/bezhanSalleh/filament-shield) currently has an implementation of this, which could be useful.

**Comment by ryangjchandler** [comment]
@intrepidws Can you not just use `mount()` and do authorization there, like a regular LW component?

**Reply by intrepidws** [reply]
Yea, you're absolutely right - that's actually what I ended up settling on. In fact, I'm now seeing that the docs DO reference that for pages. I guess my only argument would be that having a `canView()` method (or something similarly named) might be good for consistency across the framework.

**Reply by ryangjchandler** [reply]
Hm, having a `canView()` method would mean the `Page` needs to implement a base `mount()` method which would make it hard to extend and add parameters too, etc.

**Comment by Kamoba** [comment]
We can overwrite canAccess() in page:
```
    public static function canAccess(): bool
    {
        return false;
    }
```
